cmd/inferTime.go

```diff
@@ -0,0 +1,163 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"time"
+)
+
+/*
+Copyright © 2020 Jean-Marc Meessen, ON4KJM <on4kjm@gmail.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//InferTimeBlock contains the information describing a time gap
+type InferTimeBlock struct {
+	lastRecordedTime time.Time
+	nextValidTime    time.Time
+
+	//Number of records without actual time
+	noTimeCount int
+
+	//First log entry with missing date
+	logFilePosition int
+
+	//Computed time interval (in seconds)
+	deltatime int
+}
+
+//displayTimeGapInfo will print the details stored in an InferTimeBlock
+func (tb *InferTimeBlock) displayTimeGapInfo() string {
+	timeFormat := "2006-01-02 15:04"
+	buffer := ""
+	buffer = buffer + fmt.Sprintf("Last Recorded Time:                 %s\n", tb.lastRecordedTime.Format(timeFormat))
+	buffer = buffer + fmt.Sprintf("next Recorded Time:                 %s\n", tb.nextValidTime.Format(timeFormat))
+	buffer = buffer + fmt.Sprintf("Log position of last recorded time: %d\n", tb.logFilePosition)
+	buffer = buffer + fmt.Sprintf("Nbr of entries without time:        %d\n", tb.noTimeCount)
+	buffer = buffer + fmt.Sprintf("Computed interval:                  %d\n", tb.deltatime)
+
+	return buffer
+}
+
+//finalizeTimeGap makes the necessary checks and computation
+func (tb *InferTimeBlock) finalizeTimeGap() error {
+	var err error
+	err = nil
+
+	//Check that lastRecordedTime and nextValidTime are not null
+	nullTime := time.Time{}
```

Comment:
```suggestion
```

`nullTime` in that case is used as a constant or your are declaring it as a variable which is a bit misleading.
I would compare directly against `time.Time{}`, or use a pointer to `time.Time` in order to be able to check for the `nil` values in that case.

To me, a pointer: 
- Allow to pass a mutable value, which is sadly sometimes needed
- Or can be used to express the "absence" of a value, which is our case here

Reply:
Sorry. I understand your point but I don't understand the implementation. I get compiler/lint errors when trying to adapt the code thereafter. I left it as it was.

Reply:
I think I found an alternate method. `if myTime.IsZero() {....} ` seems to do the trick. Test seem to continue to work.

Reply:
That is perfectly correct. I remembered that it was possible somehow :)